src/client/mod.rs

```diff
@@ -317,14 +374,18 @@ impl WarpgrapherClient {
     /// use std::env::var_os;
```

Comment:
The documentation (a little above), should describe conditions under which errors are returned.